Sources/GRPCHTTP2TransportNIOPosix/HTTP2ClientTransport+Posix.swift

```diff
@@ -133,11 +140,43 @@ extension HTTP2ClientTransport.Posix {
     func establishConnection(
       to address: GRPCHTTP2Core.SocketAddress
     ) async throws -> HTTP2Connection {
+      #if canImport(NIOSSL)
+      let nioSSLContext: NIOSSLContext?
+      let serverHostname: String?
+      switch self.config.transportSecurity.wrapped {
+      case .plaintext:
+        nioSSLContext = nil
+        serverHostname = nil
+      case .tls(let tlsConfig):
+        do {
+          nioSSLContext = try NIOSSLContext(configuration: TLSConfiguration(tlsConfig))
```

Comment:
While making this change I realised that we're exposing the `tls` case for `TransportSecurity` even when `NIOSSL` cannot be imported. I think this is unsafe, because it means users may be able to add a `TLSConfig` to their server/client, which then simply will be ignored if the platform the code's being built on doesn't support NIOSSL. If this happens on both the server and the client, it could mean the connection is plaintext and there would be no warnings that the config wasn't used.
I've wrapped the `tls` case of `TransportSecurity` in `#if canImport(NIOSSL)` to avoid this. Let me know if you agree with it or if I'm missing something/think it should be resolved in some other way.